dev/NavigationView/NavigationView.cpp

```diff
@@ -379,10 +427,354 @@ void NavigationView::OnApplyTemplate()
     UpdateBackAndCloseButtonsVisibility();
     UpdateSingleSelectionFollowsFocusTemplateSetting();
     UpdateNavigationViewUseSystemVisual();
-    PropagateNavigationViewAsParent();
     UpdatePaneVisibility();
     UpdateVisualState();
     UpdatePaneTitleMargins();
+    SyncItemTemplates();
+}
+
+void NavigationView::UpdateRepeaterItemsSource(bool forceSelectionModelUpdate)
+{
+    auto dataSource = MenuItemsSource();
+    if (!dataSource)
+    {
+        dataSource = MenuItems();
+        UpdateSelectionForMenuItems();
+    }
+
+    // Selection Model has same representation of data regardless
+    // of pane mode, so only update if the ItemsSource data itself
+    // has changed.
+    if (forceSelectionModelUpdate){
+        m_selectionModel.Source(dataSource);
```

Comment:
>dataSource [](start = 32, length = 10)

Will any of these calls fail if dataSource is null?

Reply:
No, they shouldn't. Set itemsSource to nullptr to verify.